### ISSUE TITLE
Endless memory usage with imgui-node-editor

### ISSUE THREAD

**Issue by sean-mcmanus** [microsoft/vscode-cpptools#7085]
As I have the same or a similar problem I will use this thread to share my information.

### Description

`cpp-tools-srv` Process starts increasing its memory usage (approx. 500MB/s) till the memory and swap are full and the whole system freezes. Also I noticed, that my hard drive has a lot of IOs when it happens.

### System Information

* C/C++ Extension: v1.2.2
* VSCode Version: 1.53.2 (OSS)
Commit: 622cb03f7e070a9670c94bae1a45d78d7181fbd4
Date: 2021-02-11T21:13:58.622Z
Electron: 11.3.0
Chrome: 87.0.4280.141
Node.js: 12.18.3
V8: 8.7.220.31-electron.0
OS: ArchLinux Linux x64 5.11.2-arch1-1
* All Extensions except C/C++ disabled

### Steps to reproduce

Sadly I did not figure out what exactly causes the problem. Sometimes it happens straight away, sometimes it takes me 5 minutes to get the crash.

I am not sure if it is related to the project I am working with, but it happens in a fresh clone of this project https://github.com/thedmd/imgui-node-editor

I just have the files open in the screenshot. Switch between the files, scroll around and hover over the functions to see the Tooltips.

### C/C++: Log Diagnostics

When the memory leak occurs, the `C/C++: Log Diagnostics` command is not showing anything till I `pkill -SIGKILL cpptools-srv`

<details>
<summary>C/C++: Log Diagnostics (before crash)</summary>
<p>

```
-------- Diagnostics - 3/1/2021, 4:07:47 PM
Version: 1.2.2
Current Configuration:
{
    "name": "Linux",
    "includePath": [
        "${workspaceFolder}/**"
    ],
    "defines": [],
    "compilerPath": "/usr/bin/clang",
    "compilerArgs": [],
    "cStandard": "c11",
    "cppStandard": "c++17",
    "intelliSenseMode": "gcc-x64",
    "intelliSenseModeIsExplicit": true,
    "cStandardIsExplicit": true,
    "cppStandardIsExplicit": true,
    "compilerPathIsExplicit": true,
    "browse": {
        "path": [
            "${workspaceFolder}/**"
        ],
        "limitSymbolsToIncludedHeaders": true
    }
}
Translation Unit Mappings:
[ /home/topas/Documents/Coding/imgui-node-editor/imgui_node_editor.cpp ]:
    /home/topas/Documents/Coding/imgui-node-editor/imgui_node_editor.h
    /home/topas/Documents/Coding/imgui-node-editor/imgui_node_editor_internal.h
    /home/topas/Documents/Coding/imgui-node-editor/imgui_node_editor.cpp
[ /home/topas/Documents/Coding/imgui-node-editor/imgui_node_editor_api.cpp ]:
    /home/topas/Documents/Coding/imgui-node-editor/imgui_node_editor_api.cpp
    /home/topas/Documents/Coding/imgui-node-editor/imgui_node_editor.h
    /home/topas/Documents/Coding/imgui-node-editor/imgui_node_editor_internal.h
Translation Unit Configurations:
[ /home/topas/Documents/Coding/imgui-node-editor/imgui_node_editor.cpp ]:
    Process ID: 327386
    Memory Usage: 324 MB
    Compiler Path: /usr/bin/clang
    Includes:
        /usr/include/c++/10.2.0
        /usr/include/c++/10.2.0/x86_64-pc-linux-gnu
        /usr/include/c++/10.2.0/backward
        /usr/local/include
        /usr/lib/clang/11.1.0/include
        /usr/include
        /home/topas/Documents/Coding/imgui-node-editor/external/imgui
        /home/topas/Documents/Coding/imgui-node-editor/build/application/source
        /home/topas/Documents/Coding/imgui-node-editor
    Standard Version: c++17
    IntelliSense Mode: linux-clang-x64
    Other Flags:
        --clang
        --clang_version=110100
[ /home/topas/Documents/Coding/imgui-node-editor/imgui_node_editor_api.cpp ]:
    Process ID: 339141
    Memory Usage: 209 MB
    Compiler Path: /usr/bin/clang
    Includes:
        /usr/include/c++/10.2.0
        /usr/include/c++/10.2.0/x86_64-pc-linux-gnu
        /usr/include/c++/10.2.0/backward
        /usr/local/include
        /usr/lib/clang/11.1.0/include
        /usr/include
        /home/topas/Documents/Coding/imgui-node-editor/external/imgui
        /home/topas/Documents/Coding/imgui-node-editor/build/application/source
    Standard Version: c++17
    IntelliSense Mode: linux-clang-x64
    Other Flags:
        --clang
        --clang_version=110100
Total Memory Usage: 533 MB

------- Workspace parsing diagnostics -------
Number of folders and files enumerated: 31612
Number of files discovered (not excluded): 29788
```
</p>
</details>  

### Screenshots

![Bpytop](https://user-images.githubusercontent.com/17766093/109511281-2c836a80-7aa3-11eb-9aa7-19654f39984a.png)

Process Explorer:
![ProcessExplorer](https://user-images.githubusercontent.com/17766093/109511319-360cd280-7aa3-11eb-8034-f72bc6dfe7ad.png)

Language Server Log (notice the ToolTip Window in Line 739 is only saying "Loading...")
![LanguageServer](https://user-images.githubusercontent.com/17766093/109511329-386f2c80-7aa3-11eb-9a7d-d873ba79d56a.png)

_Originally posted by @ToppDev in https://github.com/microsoft/vscode-cpptools/issues/6922#issuecomment-788014444_

**Comment by sean-mcmanus** [comment]
Okay -- I got the repro. Investigating tomorrow...

**Comment by sean-mcmanus** [comment]
Thanks for reporting this. The bug repros in VS as well. I've filed a bug against VS but it doesn't appear on their website yet so I messaged the team directly.

**Comment by sean-mcmanus** [comment]
The issue is at https://developercommunity.visualstudio.com/t/vcpkgsrvexe-crashes-from-an-out-of-memory-with-img/1367635 , but I'm not yet if it's publicly accessible.

**Comment by sean-mcmanus** [comment]
@ToppDev Fixed with https://github.com/microsoft/vscode-cpptools/releases/tag/1.3.0-insiders .

**Comment by ToppDev** [comment]
Thank you very much

**Comment by michelleangela** [comment]
Fix is released in the 1.3.0 release version https://github.com/microsoft/vscode-cpptools/releases/tag/1.3.0
